tests/test_cmake_build/main.cpp

```diff
@@ -2,5 +2,6 @@
 namespace py = pybind11;
 
 PYBIND11_MODULE(test_cmake_build, m) {
+    m.set_gil_not_used();
```

Comment:
General question: is there an inverse? Is the there some way to say "I'm wrapping a library that is single threaded (like an old C library) so don't warn about requiring the GIL in free-threaded builds (and keep the GIL even if GIL=0), or should users manually lock?

Reply:
We can change `set_gil_not_used()` to take an argument, but there isn't really an inverse. [`PyUnstable_Module_SetGIL`](https://docs.python.org/3.13/c-api/module.html#c.PyUnstable_Module_SetGIL) takes two options: [`Py_MOD_GIL_USED`](https://docs.python.org/3.13/c-api/module.html#c.Py_mod_gil.Py_MOD_GIL_USED) and [`Py_MOD_GIL_NOT_USED`](https://docs.python.org/3.13/c-api/module.html#c.Py_mod_gil.Py_MOD_GIL_NOT_USED), but passing `Py_MOD_GIL_USED` has the same effect as not calling the function all.

In other words, there's no way to say "keep the GIL even if `PYTHON_GIL=0`". The environment variable always overrides the module behavior.

Reply:
No need to change it unless there was a way to set it without the warning.